app/controllers/application_controller.rb

```diff
@@ -37,8 +37,12 @@ def routing_error
       when Common::Exceptions::BaseError
         exception
       when Common::Client::Errors::BackendServiceError
-        additional_attributes = { source: exception.developer_message, meta: exception.as_json }
-        Common::Exceptions::ClientError.new(exception.message, additional_attributes)
+        additional_attributes = {
+          detail: exception.message,
+          source: exception.developer_message,
+          meta: exception.as_json
+        }
+        Common::Exceptions::ClientError.new(nil, additional_attributes)
```

Comment:
Sure, BackendServiceError makes sense.

I think the problem I'm having with `Common::Client::Errors` and `Common::Exceptions` is that I still don't quite understand what they're intended to do. I think I'm starting to see that `Common::Client::Errors` is meant to handle errors arising from communication with the backend APIs while `Common::Exceptions` manage issues coming from the Rails app itself. If that's the case, then maybe the problem is solved by replacing the confusing word `Client` with something more specific to the problem these classes are addressing?